nixos/modules/system/boot/stage-2-init.sh

```diff
@@ -54,7 +54,7 @@ if [ ! -e /proc/1 ]; then
 fi
 
 
-if [ "${IN_NIXOS_SYSTEMD_STAGE1:-}" = true ]; then
+if [ "${IN_NIXOS_SYSTEMD_STAGE1:-}" = true ] || [ ! -z "$container" ] ; then
```

Comment:
```suggestion
if [ "${IN_NIXOS_SYSTEMD_STAGE1:-}" = true ] || [ -n "$container" ] ; then
```

Not empty instead of negating empty

Reply:
thank you for the review! I've pushed the changes you requested.